.github/workflows/build.yml

```diff
@@ -1,78 +1,93 @@
-on: [push, pull_request]
-
 name: build
 
+on: [push, pull_request]
+
 jobs:
   build:
-    name: Lints and Test
+    name: Build
     strategy:
+      fail-fast: false
       matrix:
         platform: [ubuntu-latest, macos-latest, windows-latest]
         toolchain: [stable]
     runs-on: ${{ matrix.platform }}
 
     steps:
-      - name: Checkout sources
+      - name: Checkout Sources
         uses: actions/checkout@v2
 
-      - name: Install Rust toolchain
+      - name: Cache Dependencies & Build Outputs
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Install Rust Toolchain
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: ${{ matrix.toolchain }}
           override: true
           components: rustfmt, clippy
 
-      - name: Run cargo fmt
+      - name: Check Code Format
         uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: --all -- --check
 
-      - name: Run cargo clippy
+      - name: Code Lint
         uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --all --all-targets -- -D warnings
+          args: --all-targets --all-features --workspace -- -D warnings
 
-      - name: Run cargo test
+      - name: Test
         uses: actions-rs/cargo@v1
         with:
           command: test
+          args: --all-features --workspace
 
   ensure_no_std:
     name: Ensure no_std
     runs-on: ubuntu-latest
+
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions-rs/toolchain@v1
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+
+      - name: Install Rust Toolchain
+        uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: nightly
+          toolchain: stable
```

Comment:
Stable cannot be used until 1.50 is released. When it is released, the new resolver must be used, see 
https://github.com/rust-lang/cargo/blob/master/src/doc/src/reference/resolver.md#resolver-versions for how to do it. I suggest using nightly for now.

Reply:
Thanks!